cognite/client/config.py

```diff
@@ -244,7 +247,7 @@ def load(cls, config: dict[str, Any] | str) -> ClientConfig:
         """
         loaded = load_resource_to_dict(config)
 
-        if isinstance(loaded["credentials"], CredentialProvider):
+        if not loaded["credentials"] or isinstance(loaded["credentials"], CredentialProvider):
```

Comment:
Same comment here, if Credentials are required, let's always pass the content to the load method (except if already correct (sub)type)